theano/tensor/nlinalg.py

```diff
@@ -726,3 +726,37 @@ def norm(x, ord):
             raise ValueError(0)
     elif ndim > 2:
         raise NotImplementedError("We don't support norm witn ndim > 2")
+
+
+class LogAbsDet(Op):
+    """Computes the logarithm of absolute determinant of a square
+    matrix M, log(abs(det(M))), on CPU. Avoids det(M) overflow/
+    underflow.
+
+    TODO: add GPU code!
+    """
+    def make_node(self, x):
+        x = theano.tensor.as_tensor_variable(x)
+        o = theano.tensor.scalar(dtype=x.dtype)
+        return Apply(self, [x], [o])
+
+    def perform(self, node, inputs, outputs):
+        try:
+            (x,) = inputs
+            (z,) = outputs
+            s = numpy.linalg.svd(x, compute_uv=False)
+            log_abs_det = numpy.sum(numpy.log(numpy.abs(s)))
```

Comment:
Yeah, I have no idea what the gradient of SVD would be :) grad of log abs det is very simple on the other hand, so maybe it's better to leave this out of the theano optimizations...

OK didn't know there's no GPU op for MatrixInverse... so step 1 would now be to get LogAbsDet work on GPU? Is it possible to use the skcuda.linalg.svd directly? (I have no idea how the make_thunk works...). So not sure if I should try to do the GPU SVD here directly or in the SVD op? Damn it's still a bit confusing :)